staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -1402,3 +1424,30 @@ func buildOpenAPIModelsForApply(staticOpenAPISpec map[string]*spec.Schema, crd *
 	}
 	return mergedOpenAPI.Components.Schemas, nil
 }
+
+type cached[T any] struct {
```

Comment:
This may have some doc explaining how it works, it seems it executes a Fn and keeps the value once executed, that will be returned by subsequent executions

Reply:
Based on that description, I don't see how this is different from `sync.Once`? or [`cached.NewStaticSource`](https://github.com/kubernetes/kube-openapi/blob/master/pkg/cached/cache.go#L260-L266). Except maybe that this is going to be called again if it fails. I think that's a reasonable use-case to add to to [cached](https://github.com/kubernetes/kube-openapi/blob/master/pkg/cached/).

Reply:
`cached.NewStaticSource` makes sense.

Btw, the cached package has to move elsewhere. It has nothing to do with kube-openapi and will be used for non-openapi stuff.

Reply:
The result type feels clunky. I would very much prefer `Get() (T, string, error)`. That would feel much more like idiomatic.